test/test_iterdatapipe.py

```diff
@@ -344,6 +344,11 @@ def test_header_iterdatapipe(self) -> None:
         header_dp = source_dp.header(100)
         self.assertEqual(list(range(5)), list(header_dp))
 
+        # Functional Test: ensure the source is not modified if limit is set to None
```

Comment:
To me it seems as if test_header_iterdatapipe() contains a whole test suite comprised of multiple tests. Imho there should be one test for each case and one (logical) assertion per test.

Maybe this can be improved in a separate PR so that we don't get scope creep here @ejguan?

Reply:
Sounds reasonable to me. And, I feel it's even better to add a directory and split this single test file into a few categories like `archive_datapipe.py`, `augmenting_datapipe.py`, etc.
DataPipe are roughly categorized in https://pytorch.org/data/beta/index.html

cc: @NivekT WDYT about the test suite

Reply:
Agrees with the concerns here. I think we should refactor all the DataPipe unit tests into a better format at some point... Open to suggestions but probably beyond the scope of this PR